### PR TITLE
PXB-2491 Document keyring component (8.0)

### PR DESCRIPTION
**Pull request by patrickbirch**
Changes to be committed:
	modified:   source/advanced/encrypted_innodb_tablespace_backups.rst

### CODE REVIEW

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst [hunk 1]
```diff
@@ -52,64 +47,56 @@ message:
...
-option. In case the keyring has been rotated, you need to restore the keyring
-which was used to take and prepare the backup.
+option. In case the keyring has been rotated, you must restore the keyring used when the backup was  taken and prepared.
 
 Using ``keyring_vault`` plugin
 ------------------------------
 
 The support for encrypted InnoDB tablespace backups with ``keyring_vault`` has been
-implemented in |Percona XtraBackup| 2.4.11.  Keyring vault plugin settings are
+implemented in |Percona XtraBackup| 2.4.11. Keyring vault plugin settings are
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst]: why do we mention 2.4 in the 8.0 documentation ?

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst [hunk 2]
```diff
@@ -136,12 +122,67 @@ The backup is now prepared and can be restored with the :option:`--copy-back` op
...
+|xtrabackup| reads the keyring_file component configuration from ``xtrabackup_component_keyring_file.cnf``. You must specify the keyring_file data path if the ``keyring-file-data`` is not located in the attribute ``PATH`` from the xtrabackup_component_keyring_file.cnf. 
+
+The following is an example of adding the location for the keyring-file-data:
+
+.. sourcecode:: bash
+
+   xtrabackup --prepare --target-dir=/data/backup \ 
+   --keyring-file-data=/var/lib/mysql-keyring/keyring
+
+.. note:: |xtrabackup| attempts to read ``xtrabackup_component_keyring_file.cnf``. You can assign another keyring file component by passing the ``--component-keyring-file-config`` option.
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst]:
```suggestion
.. note:: |xtrabackup| attempts to read ``xtrabackup_component_keyring_file.cnf``. You can assign another keyring file component configuration by passing the ``--component-keyring-file-config`` option. 
```

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst [hunk 3]
```diff
@@ -136,12 +122,67 @@ The backup is now prepared and can be restored with the :option:`--copy-back` op
...
+The backup is prepared. To restore the backup use the ``--copy-back`` option. If the keyring has been rotated, you must restore the specific keyring used to take and prepare the backup.
+
+
 Incremental Encrypted InnoDB tablespace backups with ``keyring_file``
 ---------------------------------------------------------------------
 
 The process of taking incremental backups with InnoDB tablespace encryption is
 similar to taking the :ref:`xb_incremental` with unencrypted tablespace.
 
+.. note:: The ``keyring-file`` component should not used in production or for regulatory compliance.
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst:197]: very good point to have in the docs.

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]
few comments

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst (hunk 1 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst]: removed.

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst (hunk 2 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst]: fixed.

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst (hunk 3 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst:197]: Agreed.

**Review by altmannmarcelo — APPROVED** [review]
LGTM

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst [hunk 4]
```diff
@@ -136,12 +121,67 @@ The backup is now prepared and can be restored with the :option:`--copy-back` op
...
+Using the ``keyring_file`` component
+-------------------------------------
+
+A component is not loaded with the ``--early_plugin_load`` option. The server uses a manifest to load the component and the component has its own configuration file. See `component installation <https://dev.mysql.com/doc/refman/8.0/en/keyring-component-installation.html>`__ for more information.
+
+The following is an example of a manifest and a configuration file:
+
+.. sourcecode::
+
+   ./bin/mysqld.my: { "components": "file://component_keyring_file" }
```
Comment (mchawla16) [inline storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst]: Note: This is valid for global manifest and global configuration, when mysql tarball is used. Probably format it better, refer below links for configuration, when using packages
https://dev.mysql.com/doc/refman/8.0/en/keyring-component-installation.html 
https://dev.mysql.com/doc/refman/8.0/en/keyring-file-component.html

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst [hunk 5]
```diff
@@ -136,12 +121,67 @@ The backup is now prepared and can be restored with the :option:`--copy-back` op
...
+/lib/plugin/component_keyring_file.cnf:{ "path": "/var/lib/mysql-keyring/keyring_file", "read_only": false }
```
Comment (mchawla16) [inline storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst]: Format the above command as json code. This is global configuration, when tarball is used. Please refer to https://dev.mysql.com/doc/refman/8.0/en/keyring-file-component.html for configuration in package.

**Review by mchawla16 — APPROVED** [review]

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst (hunk 5 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst]: Fixed and added.

File: storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst (hunk 4 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/encrypted_innodb_tablespace_backups.rst]: fixed and added.

**Pull request merged**